ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -154,149 +241,511 @@ class Pagination {
+     * The container element for the next page button.
+     */
+    public renderNextButton(container: HTMLElement): void {
+        this.nextButton = makeHTMLElement('button', {
+            innerHTML: '&rsaquo;',
+            className: Globals.getClassName('pgButton')
+        }, container);
+        this.nextButton.title = this.lang.nextPage as string;
+
+        this.nextButton.addEventListener('click', (): void => {
+            void this.goToPage(this.currentPage + 1);
+        });
 
         this.setButtonState(
             this.nextButton,
-            (originalDataTable?.rowCount || 0) <= this.itemsPerPage
+            this.currentPage >= this.totalPages
         );
     }
 
     /**
-     * Call modifier to replace items with new ones.
+     * Render the last page button.
      *
-     * @param isNextPage
-     * Declare prev or next action triggered by button.
-     * @returns
+     * @param container
+     * The container element for the last page button.
      */
-    public async updatePage(isNextPage: boolean = true): Promise<void> {
-        const pg = this;
-        const grid = pg.grid;
-        const originalDataTable = grid.dataTable;
-        const { beforePageChange, afterPageChange } = this.options.events || {};
+    public renderLastButton(container: HTMLElement): void {
+        this.lastButton = makeHTMLElement('button', {
+            innerHTML: '&raquo;',
+            className: Globals.getClassName('pgButton')
+        }, container);
+        this.lastButton.title = this.lang.lastPage as string;
+
+        this.lastButton.addEventListener('click', (): void => {
+            void this.goToPage(this.totalPages);
+        });
+
+        this.setButtonState(
+            this.lastButton,
+            this.currentPage >= this.totalPages
+        );
+    }
+
+    /**
+     * Render page number buttons with ellipsis.
+     *
+     * @param container
+     * The container element for the page number buttons.
+     */
+    public renderPageNumbers(container: HTMLElement): void {
+        this.pageNumbersContainer = makeHTMLElement('div', {
+            className: Globals.getClassName('pgPageButton')
+        }, container);
 
-        if (!originalDataTable || !this.prevButton || !this.nextButton) {
+        this.updatePageNumbers();
+    }
+
+    /**
+     * Update page number buttons based on current page and total pages.
+     */
+    public updatePageNumbers(): void {
+        if (!this.pageNumbersContainer) {
             return;
         }
 
-        // Event trigger before page change, defined by user
-        if (beforePageChange) {
-            fireEvent(
-                this,
-                'beforePageChange',
-                {
-                    currentPage: pg.currentPage
-                },
-                beforePageChange
+        // Clear existing page numbers
+        this.pageNumbersContainer.innerHTML = '';
+
+        const maxPageNumbers = this.options.controls?.pageButtons?.count || 5;
```

Comment:
I'd move `5` to a const indicating what it stands for (perhaps outside of this method).